source/bolts/experimental/refraction.d

```diff
@@ -0,0 +1,684 @@
+/++
+This module helps building functions from other functions.
+
+ It is sometimes necessary to create a function which is an exact copy of
+ another function. Or sometimes it is necessary to introduce a few variations,
+ while carrying all the other aspects. Because of function attributes,
+ parameter storage classes and user-defined attributes, this requires building
+ a string mixin. In addition, the mixed-in code must refer only to local names,
+ if it is to work across module boundaires. This problem and its solution are
+ by Adam D. Ruppe in a Tip of the Week, available here:
+ https://stackoverflow.com/questions/32615733/struct-composition-with-mixin-and-templates/32621854#32621854
+
+ This module facilitates the  creation of such mixins.
+
+ +/
+
+module bolts.experimental.refraction;
+
+import std.array;
+import std.format;
+import std.meta;
+import std.algorithm.iteration : map;
+import std.range : iota;
+import std.traits : functionAttributes, FunctionAttribute;
+
+// Do not require caller module to import 'std.traits'. Instead use our own
+// aliases in mixtures.
+alias ReturnType = std.traits.ReturnType;
+alias Parameters = std.traits.Parameters;
+
+template ParameterAttribute(alias F, int i, int j) {
```

Comment:
UNdocumented public name

Reply:
Also moved to bottom of module so it does not appear before the main stuff in the doc.